tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -66,21 +67,193 @@ async def test_delayed_assignment_still_errors(dut):
+    await int_overflow_test(dut.stream_in_data, value)
+
+
+@cocotb.test()
+async def test_int_8bit_underflow(dut):
+    """Test 8-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data))
+    await int_overflow_test(dut.stream_in_data, value)
+
+
+@cocotb.test()
+async def test_int_32bit(dut):
+    """Test int access to 32-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_dword))
+    await int_values_test(dut.stream_in_data_dword, values)
+
+
+@cocotb.test()
+async def test_int_32bit_overflow(dut):
+    """Test 32-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_dword))
+    await int_overflow_test(dut.stream_in_data_dword, value)
+
+
+@cocotb.test()
+async def test_int_32bit_underflow(dut):
+    """Test 32-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_dword))
+    await int_overflow_test(dut.stream_in_data_dword, value)
+
+
+@cocotb.test()
+async def test_int_39bit(dut):
+    """Test int access to 39-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_39bit))
+    await int_values_test(dut.stream_in_data_39bit, values)
+
+
+@cocotb.test()
+async def test_int_39bit_overflow(dut):
+    """Test 39-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_39bit))
+    await int_overflow_test(dut.stream_in_data_39bit, value)
+
+
+@cocotb.test()
+async def test_int_39bit_underflow(dut):
+    """Test 39-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_39bit))
+    await int_overflow_test(dut.stream_in_data_39bit, value)
+
+
+@cocotb.test()
+async def test_int_64bit(dut):
+    """Test int access to 64-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_wide))
+    await int_values_test(dut.stream_in_data_wide, values)
+
+
+@cocotb.test()
+async def test_int_64bit_overflow(dut):
+    """Test 64-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_wide))
+    await int_overflow_test(dut.stream_in_data_wide, value)
+
+
+@cocotb.test()
+async def test_int_64bit_underflow(dut):
+    """Test 64-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_wide))
+    await int_overflow_test(dut.stream_in_data_wide, value)
+
+
+@cocotb.test()
+async def test_int_128bit(dut):
+    """Test int access to 128-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_dqword))
+    await int_values_test(dut.stream_in_data_dqword, values)
+
+
+@cocotb.test()
+async def test_int_128bit_overflow(dut):
+    """Test 128-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_dqword))
+    await int_overflow_test(dut.stream_in_data_dqword, value)
+
+
+@cocotb.test()
+async def test_int_128bit_underflow(dut):
+    """Test 128-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_dqword))
+    await int_overflow_test(dut.stream_in_data_dqword, value)
```

Comment:
This should be a `TestFactory` or something similar. `dut` can be accessed via `cocotb.top` from the module scope now.